File: typescript/src/problem0002/linkedList.ts
export default class ListNode {
  val: number;
  next: ListNode | null;

  constructor(val: number = 0, next: ListNode | null = null) {
    this.val = val;
    this.next = next;
  }
}

export const int2List = (nums: number[]): ListNode | null => {
  if (nums.length === 0) {
    return null;
  }

  let head = null;

  for (const num of nums.reverse()) {
    let node = new ListNode(num);
    node.next = head;
    head = node;
  }

  return head;
};

File: golang/src/problem0002/linkedList.go
package problem0002

type ListNode struct {
	Val  int
	Next *ListNode
}

func int2List(nums []int) *ListNode {
	if len(nums) == 0 {
		return nil
	}

	var head *ListNode

	for i := len(nums) - 1; i >= 0; i-- {
		node := &ListNode{Val: nums[i]}
		node.Next = head
		head = node
	}

	return head
}
